spring-cloud-aws-cloudmap/src/main/java/org/springframework/cloud/aws/cloudmap/CloudMapUtils.java

```diff
@@ -0,0 +1,457 @@
+import com.amazonaws.services.servicediscovery.model.ServiceAlreadyExistsException;
+import com.amazonaws.services.servicediscovery.model.ServiceFilter;
+import com.amazonaws.services.servicediscovery.model.ServiceSummary;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cloud.aws.cloudmap.discovery.CloudMapServiceInstance;
+import org.springframework.cloud.aws.cloudmap.exceptions.CreateNameSpaceException;
+import org.springframework.cloud.aws.cloudmap.exceptions.CreateServiceException;
+import org.springframework.cloud.aws.cloudmap.exceptions.MaxRetryExceededException;
+import org.springframework.cloud.aws.cloudmap.model.discovery.CloudMapDiscoveryProperties;
+import org.springframework.cloud.aws.cloudmap.model.registration.CloudMapRegistryProperties;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.core.env.Environment;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.StringUtils;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * Uses Fargate Metadata URL to retrieve IPv4 address and VPC ID to register instances to
+ * cloudmap.
+ *
+ * @author Hari Ohm Prasath
+ * @since 2.3.2
+ */
+public enum CloudMapUtils {
+
+	INSTANCE;
+	public final String NAMESPACE_ID = "NAMESPACE_ID";
+	public final String SERVICE_ID = "SERVICE_ID";
+	public final String SERVICE_INSTANCE_ID = "SERVICE_INSTANCE_ID";
+	public final String IPV_4_ADDRESS = "IPV4_ADDRESS";
+	final RestTemplate REST_TEMPLATE = new RestTemplate();
+	final Logger LOGGER = LoggerFactory.getLogger(CloudMapUtils.class);
+	final ObjectMapper JSON_MAPPER = new ObjectMapper();
+	final String VPC_ID = "VPC_ID";
+	final String AWS_INSTANCE_IPV_4 = "AWS_INSTANCE_IPV4";
+	final String REGION = "REGION";
+	private static final String SUBMITTED = "SUBMITTED";
+	private static final String PENDING = "PENDING";
+	private static final int MAX_POLL = 30;
+
+	private AmazonEC2 ec2Client;
+
+	/**
+	 * Uses ECS Fargate metadata URL to fetch all the required details around IP address
+	 * and VpcID to register instances to cloudmap service.
+	 *
+	 * @return map containing ip address and vpcid
+	 */
+	public Map<String, String> getRegistrationAttributes() {
+		Map<String, String> attributes = new HashMap<>();
+		try {
+			ResponseEntity<String> metaDataResponse = REST_TEMPLATE
+				.getForEntity(System.getenv("ECS_CONTAINER_METADATA_URI_V4") + "/task", String.class);
+			JsonNode root = JSON_MAPPER.readTree(metaDataResponse.getBody());
+			JsonNode jsonNode = root.get("Containers").get(0).get("Networks").get(0);
```

Comment:
is this always true? I mean the first item in the container list and the first network? In general I think is ok but just double checking

Reply:
Good catch. I have created two methods which will be picked up based on the `DEPLOYMENT_PLATFORM` value `ECS`

* `getEcsRegistrationAttributes` - Used for ECS
* `getEksRegistrationAttributes` - Used for EKS